extensions/sql-migration/src/wizard/integrationRuntimePage.ts

```diff
@@ -648,20 +674,43 @@ export class IntergrationRuntimePage extends MigrationWizardPage {
 				const state = migrationService.properties.integrationRuntimeState;
 				if (state === 'Online') {
 					await this._dmsStatusInfoBox.updateProperties(<azdata.InfoBoxComponentProperties>{
-						text: constants.SERVICE_READY(serviceName, nodeNames.join(', '), true),
+						text: constants.SERVICE_READY_WITHOUT_NODENAMES(serviceName),
 						style: 'success'
 					});
 				} else {
 					await this._dmsStatusInfoBox.updateProperties(<azdata.InfoBoxComponentProperties>{
-						text: constants.SERVICE_NOT_READY(serviceName, true),
+						text: constants.SERVICE_NOT_READY(serviceName, false),
 						style: 'error'
 					});
 				}
 
 				// exit if new call has started
 				if (callSequence !== this._lastIn) { return; }
 
-				await utils.refreshAuthenticationKeyTable(this._view, this._authKeyTable, account, subscription, resourceGroup, location, migrationService);
+				// populate the table with data
+				await utils.refreshIntegrationRuntimeTable(this._view, this._integrationRuntimeTable, migrationServiceMonitoringStatus);
+
+				// based on the data, enable or disable configure ir button
+				if (migrationServiceMonitoringStatus.nodes.length === 4 &&
+					this.atleastOneNodeOnline(migrationServiceMonitoringStatus.nodes)) {
+					this._configureIRButton.enabled = false;
+				}
+				else {
+					this._configureIRButton.enabled = true;
+				}
+
+				// if the versions are mismatched, show a warning
+				if (migrationServiceMonitoringStatus?.nodes.length === 1) {
+					let nodeversion = migrationServiceMonitoringStatus.nodes[0].version;
+					migrationServiceMonitoringStatus.nodes.forEach(async node => {
+						if (node.version !== nodeversion && node.status === 'Online') {
```

Comment:
The code inside forEach is using async functions. Note that forEach doesn't work well with async/await due to the way it handles the return values of the callback function. Consider using for...of loop for asynchronous operations.

Like
for (const node of migrationServiceMonitoringStatus.nodes) {
}